### DISCUSSION TITLE
Is there any way to support multi tenancy while integration Quarkus with quarkus-messaging-rabbitmq.

### DISCUSSION THREAD

**Discussion by nitinty** [quarkusio/quarkus#42197] [Q&A]
Is there any way to support multi tenancy while integrating Quarkus application with quarkus-messaging-rabbitmq ?

Please assist.

[a comment by an automated account is not shown]

**Comment by nitinty** [comment]
@geoand @cescoffier

**Reply by geoand** [reply]
This is one for @ozangunalp and @cescoffier

**Comment by ozangunalp** [comment]
If by multi-tenancy you mean to configure different brokers per channel, you can do that. 
Instead of configuring `rabbitmq-host` and `rabbitmq-port` globally, you can configure host and port per channel with `mp.messaging.incoming.[channel].host` and `mp.messaging.incoming.[channel].port`.

**Reply by nitinty** [reply]
Thanks for your reply !

Actually our requirement is, we have one application which serves multiple clients so when I publish/consume messages that should be publish/consume to/from that specific client's queue based on the **X-TENANT-ID** which we are getting in request headers.
